src/container.tsx

```diff
@@ -45,7 +45,7 @@ export class Container extends React.Component<Properties> {
   componentDidUpdate(prevProps: Properties) {
     const { route, provider } = this.props;
 
-    if (route && (route !== prevProps.route)) {
+    if (route && (route !== prevProps.route) || (provider !== prevProps.provider)) {
```

Comment:
looking at this again, i think this is doing more than we want. pretty sure all we want here is to refresh if the network/chain changes, not the provider. there is any number of reasons that the provider could change, and we don't necessarily want to reload for those reasons. let's just pass the chain in to the app, and refresh based on that. (i have a PR that passes chain in.)